File: Intf/Model/EmpresaFornecedorModel.cs
using System;

namespace Intf
{
	public class EmpresaFornecedorModel : IModel
	{
		public Guid Id { get; set; }
		public DateTime? CreatedAt { get; set; }
		public DateTime? UpdatedAt { get; set; }
		public string? CreatedBy { get; set; }
		public string? UpdatedBy { get; set; }

		public Guid EmpresaId { get; set; }
		public Guid FornecedorId { get; set; }

		public EmpresaModel Empresa { get; set; } = null!;
		public FornecedorModel Fornecedor { get; set; } = null!;
	}
}

File: frontend/src/app/models/empresa-fornecedor.model.ts
import { Empresa } from './empresa.model';
import { Fornecedor } from './fornecedor.model';

export interface EmpresaComFornecedores {
  empresa: Empresa;
  fornecedores: Fornecedor[];
}

export interface FornecedorComEmpresas {
  fornecedor: Fornecedor;
  empresas: Empresa[];
}

export interface EmpresaFornecedor {
  id: string;
  empresaId: string;
  fornecedorId: string;
}
